cityofaustin/atd-moped
# Normalize project status (part 2)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/7910

See also #871, #873.

This PR continues the project status normalization by removing dependencies on the `moped_status` table, and clearing references to `moped_project.current_status` and `moped_project.current_phase`, particularly in the `ProjectPhases` component which does most of the heavy lifting for setting status and phase.

Once these changes are merged, we can safely delete the `moped_status` table.

Summary of changes:
- Changes the default status badge from "Active" to "Unknown". The default badge will render if a project does not have a current `moped_proj_phase`. 
- Removes the "Cancel" and "Put on hold" project menu actions - users should instead add a "Canceled" or "On hold" phase to a project
- Removes mutations that set `current_phase`, `current_status`, and `status_id`
- Causes "Canceled" projects to be included in the project list. 

## Testing
**URL to test:** Local


**Steps to test:**
1. In the project list view, notice that 5 projects have a status badge of "Unknown" - these projects do not have a current `moped_proj_phase`.
2. Edit "Avenue G Healthy Street" (ID 159) - add a construction phase with a start date and make the phase current
3. Confirm the status badge updates accordingly
4. Back in the project list view, notice that the status badge is rendered and the "Construct start" column is populated.
5. Create a new signal project. Confirm it has a status badge of "potential".
6. Delete this project's potential phase. Confirm the badge changes to "Unknown".
7. On the Dashboard tab, verify this project is on the "Following" tab of your project list
8. Click on your project's status badge and use the modal to add a canceled phase - make it current
9. Confirm the status badge is red with the label "Canceled".
10. Back in the main project list, verify your project appears with the red canceled label.
11. Use the advanced search to search on the `Status` field `is` `canceled`. Verify the results are correct.
12. Export the project list and verify the `current_phase` and `current_phase_key` columns are populated
13. From the project summary tab, add a subproject to a project and verify the project's status badge appears correctly in the subprojects table.

---
#### Ship list
